src/Workspaces/CSharp/Portable/Indentation/CSharpIndentationService.Indenter.cs

```diff
@@ -31,75 +30,42 @@ protected override ISmartTokenFormatter CreateSmartTokenFormatter(Indenter inden
             return new CSharpSmartTokenFormatter(indenter.OptionSet, rules, indenter.Root);
         }
 
-        protected override IndentationResult GetDesiredIndentationWorker(
-            Indenter indenter, SyntaxToken token, TextLine previousLine, int lastNonWhitespacePosition)
+        protected override IndentationResult GetDesiredIndentationWorker(Indenter indenter, SyntaxToken? tokenOpt, SyntaxTrivia? triviaOpt)
+            => TryGetDesiredIndentation(indenter, triviaOpt) ??
+               TryGetDesiredIndentation(indenter, tokenOpt) ?? default;
```

Comment:
C#'s behavior is much simpler now.  Check to see if the preceding trivia (if there) shoudl be what we indent against, otherwise indent against the preceding token.